File: REST/python/Targets/register-polling-tentacle.py
import json
import requests

octopus_server_uri = 'https://your.octopus.app/api'
octopus_api_key = 'API-YOURAPIKEY'
headers = {'X-Octopus-ApiKey': octopus_api_key}

def get_octopus_resource(uri):
    response = requests.get(uri, headers=headers)
    response.raise_for_status()

    return json.loads(response.content.decode('utf-8'))


def get_by_name(uri, name):
    resources = get_octopus_resource(uri)
    return next((x for x in resources if x['Name'] == name), None)


space_name = 'Default'
environment_names = ['Development', 'Test']
target_name = 'your-target-name'
target_tentacle_thumbprint = 'your-tentacle-thumbprint'

# The subscription id is a random 20 character id (for example: 3hw9vtskv2cbfw7zvpje) that is used to queue messages from the server to the Polling Tentacle. 
# This should match the value in the Tentacle config file.
target_polling_subscription_identifier = 'your-target-subscription'

space = get_by_name('{0}/spaces/all'.format(octopus_server_uri), space_name)
environments = get_octopus_resource('{0}/{1}/environments/all'.format(octopus_server_uri, space['Id']))
environment_ids = [environment['Id'] for environment in environments if environment['Name'] in environment_names]

target = {
    'Endpoint': {
        'CommunicationStyle': 'TentacleActive',
        'Thumbprint': target_tentacle_thumbprint,
        'Uri': 'poll://{0}'.format(target_polling_subscription_identifier)
    },
    'EnvironmentIds': environment_ids,
    'Name': target_name,
    'Roles': ['your-target-role'],
    'Status': 'Unknown',
    'IsDisabled': False
}

uri = '{0}/{1}/machines'.format(octopus_server_uri, space['Id'])
response = requests.post(uri, headers=headers, json=target)
response.raise_for_status()

File: REST/Golang/Targets/RegisterPollingTentacle.go
package main

import (
	"fmt"
	"log"
	"net/url"

	"github.com/OctopusDeploy/go-octopusdeploy/octopusdeploy"
)

func main() {

	apiURL, err := url.Parse("https://YourURL")
	if err != nil {
		log.Println(err)
	}
	APIKey := "API-YourAPIKey"

	spaceName := "Default"
	hostName := "MyPollingTentacle"
	environments := []string{"Development", "Test"}
	roles := []string{"MyRole"}
	tentacleThumbprint := "PollingTentacleThumbprint"
	tentacleIdentifier := "PollingTentacleIdentifier"

	// Get the space object
	space := GetSpace(apiURL, APIKey, spaceName)

	// Creat client for space
	client := octopusAuth(apiURL, APIKey, space.ID)

	// Get the environment ids
	environmentIds := GetEnvironmentIds(client, environments)
	pollingUrl, err := url.Parse("poll://" + tentacleIdentifier + "/")
	if err != nil {
		log.Println(err)
	}

	newDeploymentTargetEndPoint := octopusdeploy.NewPollingTentacleEndpoint(pollingUrl, tentacleThumbprint)
	newDeploymentTargetEndPoint.CommunicationStyle = "TentacleActive"

	newDeploymentTarget := octopusdeploy.NewDeploymentTarget(hostName, newDeploymentTargetEndPoint, environmentIds, roles)

	machine, err := client.Machines.Add(newDeploymentTarget)
	if err != nil {
		log.Println(err)
	}

	fmt.Println(machine)
}

func octopusAuth(octopusURL *url.URL, APIKey, space string) *octopusdeploy.Client {
	client, err := octopusdeploy.NewClient(nil, octopusURL, APIKey, space)
	if err != nil {
		log.Println(err)
	}

	return client
}

func GetSpace(octopusURL *url.URL, APIKey string, spaceName string) *octopusdeploy.Space {
	client := octopusAuth(octopusURL, APIKey, "")

	spaceQuery := octopusdeploy.SpacesQuery{
		Name: spaceName,
	}

	// Get specific space object
	spaces, err := client.Spaces.Get(spaceQuery)

	if err != nil {
		log.Println(err)
	}

	for _, space := range spaces.Items {
		if space.Name == spaceName {
			return space
		}
	}

	return nil
}

func GetEnvironmentIds(client *octopusdeploy.Client, environmentNames []string) []string {
	environmentIds := []string{}

	for _, environmentName := range environmentNames {
		environment, err := client.Environments.GetByName(environmentName)
		if err != nil {
			log.Println(err)
		}

		environmentIds = append(environmentIds, environment[0].ID)
	}

	return environmentIds
}